### PR TITLE
Feat/file to board

### LINKED ISSUE #6: export board to file + read file and convert it to a board
reminder: each element in board contains
```
dict = {
    "id": str,
    "value": int,
    "asset": Path
}
```
### exporting to file
1. initialize 2d list containing 0's. must be same size as the board generated. we'll call this `placeholder` in this example
2. get the current state of the board using `board` property
3. loop through elements in `board` and `placeholder` concurrently
4. read `"id"` from an element in `board`
    - if "queen", make the value in the corresponding element in `placeholder` 1 instead of the initialized 0
    - else, leave it 0
5. now you have a `placeholder`, a 2d list containing 0's and 1's
6. just export this to a text file in this format
```
0 0 1 0
1 0 0 0
0 0 0 1
0 1 0 0
```
___
### reading from file and converting it to a board
1. read from file. this should be converted to a 2d list. we'll call this `file` in this example
2. initialize a 2d list, we'll call this `board` in this example, containing elements:
```
dict = {
    "id": str,
    "value": int,
    "asset": Path
}
```
- use _generate_board(), then board = self.board
3. loop through elements in `file` and `board` concurrently
4. read element in `file`
    - if 1, place_queen(row, column)

### PR DESCRIPTION
**Pull request by marcusandrev**

### CODE REVIEW

**Comment by marcusandrev** [comment]
The code I added on app.py is to print and check imported_boards list. 

**Format:** <visualization of tile if queen (Q) or killzone (.)> <dictionary of the tile>
`.       {'id': 'killzone', 'value': 0, 'asset': 'path/to/killzone/asset'}`
`Q       {'id': 'queen', 'value': 0, 'asset': 'path/to/valid/queen/asset'}`

image
image

- You could see that there are a total of 92 boards, which is right naman

**Comment by am-cid** [comment]
1. it's okay to use it for testing but don't track changes to `app.py` since that's where the gui will go.
2. killzone value is +1 and should increment whenever two or more queen's killzone overlap. doing it like this will make all values of every tile 0 which is not intended behavior
___
### you wouldn't have to worry about setting proper values if you do it this way:
1. use EightQueens `reset_board()` method to make `self.board` a blank board
2. iterate through the file like normal but using indices (see step 3)
3. use indices of found `Q`s to place a queen in self.board using `place_queen(row, column)`
    - this will make sure all killzones have proper values and that queens and killzones are given assets, just like if a user inputted it in interactive mode
4. after placing all queens, append `self.board` to `self.imported_boards`
5. use `reset_board()` to reset `self.board` for the next file to use

**Review by am-cid — DISMISSED** [review]
1. it's okay to use it for testing but don't track changes to `app.py` since that's where the gui will go.
2. killzone value is +1 and should increment whenever two or more queen's killzone overlap
___
### you wouldn't have to worry about setting proper values if you do it this way:
1. use EightQueens `reset_board()` method to make `self.board` a blank board
2. iterate through the file like normal but using indices (see step 3)
3. use indices of found `Q`s to place a queen in self.board using `place_queen(row, column)`
    - this will make sure all killzones have proper values and that queens and killzones are given assets, just like if a user inputted it in interactive mode
4. after placing all queens, append `self.board` to `self.imported_boards`
5. use `reset_board()` to reset `self.board` for the next file to use

___
I understand that you only need to render the board so only the assets are needed but this will serve as an additional check for imported boards. it's not implied that imported boards are the solutions (although that is what we're doing now). in order to account for incorrect solutions (for example, the user inputted their own import_dir with their own solutions), doing it like this will expose those and make `file_to_board()` more general purpose to help in rendering boards, not just solutions

**Comment by am-cid** [comment]
*I probably shouldn't have commented above as request changes, oops mb*

although if you guys want to just keep the alternate mode as just showing solutions and not what I described (a general rendering of boards in a directory), this implementation is all good. way lighter than what I suggested above, that's for sure, since it only handles assets and ids

File: algorithm/eight_queens.py [hunk 1]
```diff
@@ -8,6 +8,7 @@ def __init__(self, size: int = 8):
         self._size = size
         self._queen_positions = []
         self._board = [[{}]]
+        self.imported_boards = []
```
Comment (chaaals) [inline algorithm/eight_queens.py]: pa follow nalang ng standard natin for this
```suggestion
        self._imported_boards = []
```
then gawa ka nalang ng method to extract the data
```py
@property
def imported_boards():
     return self._imported_boards
```

File: app.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from algorithm.eight_queens import EightQueens
```
Comment (chaaals) [inline app.py]: same comment with cid, pa separate ng testing, app will only contain GUI

**Review by chaaals — CHANGES_REQUESTED** [review]
thanks @marcusandrev ! minor changes lang naman

**Review by chaaals — APPROVED** [review]

**Pull request merged**